lib/mongoid/history/options.rb

```diff
@@ -14,44 +14,44 @@ def scope
 
       def prepared
         @prepared ||= begin
+          @prepared = options.dup
```

Comment:
This looks strange. Change it to:

```ruby
return if @prepared 

@prepared = ...
```

Generally I am a bit concerned that we are now making assignment of `prepared` and modifying it all over the place. Is there a way to fix the issue without doing that change and keep gathering settings and returning modified version from functions, then calling `merge` on the results?

Reply:
I really feel like the library took a turn down a wrong path with the linked commit and I'm just trying to make the best of it. `@options` was assigned and modified all over the place already, I just switched it so the original options could be retrieved. Another direction I considered was having Options keep and expose the `original_options` and leave it modifying `@options` and using `@prepared` in the same way it does now.

Reply:
I like keeping `original_options`, feels cleaner, no?

Reply:
Also I am fine with the code as is. LMK when you want to merge (on green).

Reply:
This is an uglier changeset in the commit, but it feels a little cleaner to me to in the end to keep the mutation done by `prepared` isolated to `@prepared` instead of `@options`.

The Travis build is green, but neither time has it updated the GitHub status. 😕